File: pytools/data/data.code.cpp.server/AckRoleLoginOk.cpp
#ifndef _ACK_ROLE_LOGIN_OK_
#define _ACK_ROLE_LOGIN_OK_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class AckRoleLoginOk
{
private:
	String m_uname;


public:
	Packet Encode()
	{
		Packet packet;
		packet.WriteString(m_uname);
		packet.Encode(Msg::P_ACK_ROLE_LOGIN_OK);
		return packet;
	}


	void SetUname(String uname)
	{
		m_uname = uname;
	}

};

#endif

File: pytools/examples/csharp/server/Protocol/AckRoleLoginOk.cs
using System.Collections;
using System.Collections.Generic;


public class AckRoleLoginOk
{
	private string _uname;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteString(this._uname);
		packet.Encode(Msg.P_ACK_ROLE_LOGIN_OK);
		return packet;
	}


	public string uname
	{
		get { return this._uname; }
		set { this._uname = value; }
	}

}
